src/algorithms/meta/FilterByAssociations.h

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Simon Gardner
+
+// Filters a collection by the members of another collection
+// The first collection is the collection which provides a filter
+// The second is the one which is divided up into two collections where its elements either passed the filter or not
+// Functions need to be provided along with the collections to form the link between the data types
+// These functions are envisioned to link the objectIDs of the collection/associations but could be anything
+
+#pragma once
+
+#include <spdlog/spdlog.h>
+#include <algorithms/algorithm.h>
+#include <string>
+#include <string_view>
+
+#include "services/log/Log_service.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+   template<class FilterByObjectT, class ToFilterObjectT>
+     using FilterByAssociationsAlgorithm =  algorithms::Algorithm<
+       typename algorithms::Input<
+        typename FilterByObjectT::collection_type,
+        typename ToFilterObjectT::collection_type
+        >,
+       typename algorithms::Output<
+        typename ToFilterObjectT::collection_type,
+        typename ToFilterObjectT::collection_type
+        >
+     >;
+
+
+  template<typename FilterByObjectT,auto FilterByFunction,typename ToFilterObjectT,auto ToFilterFunction>
+  class FilterByAssociations : public FilterByAssociationsAlgorithm<FilterByObjectT,ToFilterObjectT> {
```

Comment:
At this point, there is nothing in this that requires ToFilterObjectT to be an association. We could come up with a simpler name
```suggestion
  class FilterMatching : public FilterMatchingAlgorithm<FilterByObjectT,ToFilterObjectT> {
```
Also I'd swap the arguments. Perhaps, it's more natural to expect the first argument to be propagated, even if in some case it will be an association.

Reply:
The argument name and ordering had evolved to be a bit unnatural, thanks. This should now be fixed.